src/components/MaanedsbeloepAvansertBeregning/hooks.ts

```diff
@@ -0,0 +1,135 @@
+import { getSelectedLanguage } from '@/context/LanguageProvider/utils'
+import { useAppSelector } from '@/state/hooks'
+import {
+  selectCurrentSimulation,
+  selectFoedselsdato,
+} from '@/state/userInput/selectors'
+import { transformUttaksalderToDate } from '@/utils/alder'
+
+import {
+  hentSumOffentligTjenestepensjonVedUttak,
+  hentSumPensjonsavtalerVedUttak,
+} from '../Pensjonsavtaler/utils'
+
+export interface Pensjonsdata {
+  alder: Alder
+  grad: number
+  afp?: number
+  pensjonsavtale: number
+  alderspensjon?: number
```

Comment:
Hvis `afp` og `alderspensjon` alltid skal være med, men kan være undefined, kan du type det slik:
```suggestion
  afp: number | undefined
  pensjonsavtale: number
  alderspensjon: number | undefined
```
Da får du typefeil hvis du glemmer en av dem.